### PR TITLE
Remove guest tags based on config

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Remove guest tags based on config

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-53248

#### Release Note
```release-note
Remove guest tags based on server configuration
```

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]

File: app/components/post_list/post/system_message/system_message.tsx [hunk 1]
```diff
@@ -221,35 +222,35 @@ const renderUnarchivedMessage = ({post, author, location, styles, intl, theme}:
...
+const renderAddGuestToChannelMessage = ({post, location, styles, intl, theme}: RenderersProps, hideGuestTags: boolean) => {
     if (!post.props.username || !post.props.addedUsername) {
         return null;
     }
 
     const username = renderUsername(post.props.username);
     const addedUsername = renderUsername(post.props.addedUsername);
 
-    const localeHolder = {
+    const localeHolder = hideGuestTags ? postTypeMessages[Post.POST_TYPES.ADD_TO_CHANNEL].one : {
```
Comment (M-ZubairAhmed) [inline app/components/post_list/post/system_message/system_message.tsx:233]: Why not change here to formattedMessage when we can ?

**Review by M-ZubairAhmed — APPROVED** [review]

File: app/components/post_list/post/system_message/system_message.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply app/components/post_list/post/system_message/system_message.tsx:233]: Discussed internally, not a priority.

**Review by abhijit-singh — APPROVED** [review]
💯

**Review by yasserfaraazkhan — APPROVED** [review]
tested on android, working as expected


WhatsApp Image 2023-06-28 at 9 29 58 PM (1)

WhatsApp Image 2023-06-28 at 9 29 58 PM

WhatsApp Image 2023-06-28 at 9 29 59 PM

**Comment by larkox** [comment]
@johndavidlugtu Friendly reminder on this review

**Review by johndavidlugtu — APPROVED** [review]
Thanks for the ping -- LGTM!

**Pull request merged**